test/Autofac.Specification.Test/Util/AsyncOnlyDisposeTracker.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Autofac Project. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Autofac.Specification.Test.Util
+{
+    public class AsyncOnlyDisposeTracker : IAsyncDisposable
+    {
+        private readonly bool _completeAsync;
+
+        public AsyncOnlyDisposeTracker(bool completeAsync = false)
+        {
+            _completeAsync = completeAsync;
+        }
+
+        public event EventHandler<EventArgs> Disposing;
+
+        public bool IsAsyncDisposed { get; set; }
+
+        public async ValueTask DisposeAsync()
+        {
+            if (_completeAsync)
+            {
+                await Task.Delay(1);
```

Comment:
Would `await Task.CompletedTask` accomplish the same thing or is the intent here to also introduce a minor timing delay to force some nature of asynchronicity into it?

Reply:
The intent here is just to force the method to be properly async, rather than completing synchronously, therefore ensuring that calling code that checks the `!vt.IsCompletedSuccessfully` flag gets invoked, and does so on a different thread.

Reply:
@alistairjevans you can use `Task.Yield()`. it doesn't involve timer queue.